src/main/scala/beam/utils/VMInfoCollector.scala

```diff
@@ -0,0 +1,57 @@
+package beam.utils
+
+import java.lang.management.ManagementFactory
+
+import javax.management.{
+  InstanceNotFoundException,
+  MBeanException,
+  MBeanServer,
+  MalformedObjectNameException,
+  ObjectName,
+  ReflectionException
+}
+
+object VMInfoCollector {
+  private val DIAGNOSTIC_COMMAND_MBEAN_OBJECT_NAME = "com.sun.management:type=DiagnosticCommand"
+
+  def apply(): VMInfoCollector = {
+    try {
+      val objectName = new ObjectName(DIAGNOSTIC_COMMAND_MBEAN_OBJECT_NAME)
+      new VMInfoCollector(objectName)
+    } catch {
+      case _: MalformedObjectNameException =>
+        throw new RuntimeException(
```

Comment:
Let's not throw away that exception, but instead, pass it to `RuntimeException` so there will be more information what went wrong.

Reply:
done